vita3k/renderer/src/gl/compile_program.cpp

```diff
@@ -105,8 +105,210 @@ static SharedGLObject compile_spirv(GLenum type, const std::vector<std::uint32_t
     return shader;
 }
 
+static bool get_shaders_cache_hashs(GLState &renderer, const char *base_path, const char *title_id) {
+    const auto shaders_hashs_path{ fs::path(base_path) / "cache/shaders" / title_id / "hashs.dat" };
+    fs::ifstream shaders_hashs(shaders_hashs_path, std::ios::in | std::ios::binary);
+    if (shaders_hashs.is_open()) {
+        renderer.shaders_cache_hashs.clear();
+        // Read size of hashes list
+        size_t size;
+        shaders_hashs.read((char *)&size, sizeof(size));
+
+        // Check version of cache
+        size_t version_size;
+        shaders_hashs.read((char *)&version_size, sizeof(version_size));
+        std::vector<char> buffer(version_size);
+        shaders_hashs.read(buffer.data(), version_size);
+        const std::string versionInFile(buffer.begin(), buffer.end());
+        if (versionInFile != "1.00") {
```

Comment:
You should make version being int variable. Read an integer then check version.

Ifstream fi;
Int version;
Fi.read(&version, sizeof(int32_t));
if (version!=1)
 .....

Reply:
i have used it for app cache,i prefer, i can set 1.01 like that, int is to limited

Reply:
done, i have move it to int